apps/api/src/utils/crypto-helpers.ts

```diff
@@ -0,0 +1,102 @@
+import * as crypto from 'crypto';
+
+export const encrypt = (
+  text: string,
+  key: string,
+  algorithm = 'AES-256-GCM' as crypto.CipherGCMTypes,
+  IV_LENGTH = 12,
+  authTagLength = 16,
+): string => {
+  if (!text || !key) {
+    throw new Error('Text and key are required.');
+  }
+
+  if (key.length !== 64) {
+    throw new Error('Key must be 32 bytes in hexadecimal.');
+  }
+
+  try {
+    const initializationVector = crypto.randomBytes(IV_LENGTH);
+    const cipher = crypto.createCipheriv(
+      algorithm,
+      Buffer.from(key, 'hex'),
+      initializationVector,
+      { authTagLength },
+    );
+
+    let encrypted = cipher.update(text, 'utf8', 'hex');
+    encrypted += cipher.final('hex');
+
+    const authenticationTag = cipher.getAuthTag();
+
+    return `${initializationVector.toString(
+      'hex',
+    )}:${encrypted}:${authenticationTag.toString('hex')}`;
+  } catch (error) {
+    throw new Error('Encryption failed.');
```

Comment:
There is no detail about caught error. Different error types should be handled and error should incapsulate in a custom exception

EncryptionError(err)